### PR TITLE
glow implement

### PR DESCRIPTION
**Pull request by bojone**
keras implement of glow model. ( https://blog.openai.com/glow/ )

it will be interesting to implement invertible model (flow model) in keras. it is a very reading-friendly version of glow.

### CODE REVIEW

**Comment by bojone** [comment]
I can not understand what's wrong in python2.7 with tensorflow backend. can you help me? @gabrieldemarmiesse

**Comment by gabrieldemarmiesse** [comment]
Your pull request is fine. There is an issue with keras' test, but it has nothing to do with your code. See #10995.

**Comment by bojone** [comment]
an simple demo: https://github.com/bojone/flow/blob/master/README.md

File: examples/celeba_glow.py [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+
+height, width = misc.imread(imgs[0]).shape[:2]
+center_height = int((height - width) / 2)
+
+img_size = 64  # for a fast try, please use img_size=32
+depth = 10  # orginal paper use depth=32
+level = 3  # orginal paper use level for 256*256 CelebA HQ
+
+
+def imread(f):
```
Comment (fchollet) [inline examples/celeba_glow.py]: Since this function does more than reading an image.

`keras.preprocessing.image.load_img` already does something similar.

I suggest instead introducing a `normalize_img` function that does `img / 256 - 0.5`, and chaining it with `load_img`

File: examples/celeba_glow.py [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+        idxs = range(shape[0])
+        return idxs[::-1]
+
+    def random_initializer(self, shape):
+        idxs = range(shape[0])
+        np.random.shuffle(idxs)
+        return idxs
+
+    def call(self, inputs):
+        num_axis = len(K.int_shape(inputs))
```
Comment (fchollet) [inline examples/celeba_glow.py]: It's called `ndim`, you can obtain it with `K.ndim(inputs)`

File: examples/celeba_glow.py [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+    def __init__(self, mode='reverse', **kwargs):
+        super(Permute, self).__init__(**kwargs)
+        self.idxs = None
+        self.mode = mode
+
+    def build(self, input_shape):
+        in_dim = input_shape[-1]
+        if self.idxs is None:
+            if self.mode == 'reverse':
+                self.idxs = self.add_weight(
```
Comment (fchollet) [inline examples/celeba_glow.py]: These are not weights that are updated during training, thus you do not need to create them with `add_weight`. You could simply create tensors and set them as layer attributes (make sure the random state is seeded so you get the same weights when loading a saved model).

File: examples/celeba_glow.py (hunk 1 above)
Comment (bojone) [reply examples/celeba_glow.py]: I am not very familiar with keras's utils part. because it does not appear in https://keras.io 
I think they are some simple operations, so I do that by myself.

File: examples/celeba_glow.py (hunk 3 above)
Comment (bojone) [reply examples/celeba_glow.py]: I don't agree this. self.idxs is actually a (non-trainable but random) weight for this layer, so I use `add_weight` to ask model save this weights. I think it is a standard usage of keras custom layer.

**Comment by bojone** [comment]
@fchollet actually, the code is too long compared with the other example codes. And I also put it in my personal github. However, I think this maybe the first invertible model implement of keras, it will be a good example to show keras can do almost everything you want succinctly and fairily, comparing with the tensorflow implement and pytorch implement.
